fast64_internal/sm64/sm64_objects.py

```diff
@@ -1135,14 +1135,38 @@ class WarpNodeProperty(bpy.types.PropertyGroup):
 	warpFlagEnum : bpy.props.EnumProperty(name = 'Warp Flags Value', default = 'WARP_NO_CHECKPOINT', items = enumWarpFlag)
 	instantOffset : bpy.props.IntVectorProperty(name = 'Offset',
 		size = 3, default = (0,0,0))
+	instantWarpObject1 : bpy.props.PointerProperty(name = 'Object 1', type = bpy.types.Object)
+	instantWarpObject2 : bpy.props.PointerProperty(name = 'Object 2', type = bpy.types.Object)
+	useOffsetObjects : bpy.props.BoolProperty(name = 'Use Offset Objects', default = False)
 
 	expand : bpy.props.BoolProperty()
 
+	def calc_offsets_from_objects(self):
+		if self.instantWarpObject1 is None or self.instantWarpObject2 is None:
+			raise PluginError(f'Warp Start and Warp End in Warp Node {self.warpID} must have objects selected.')
+		difference = self.instantWarpObject2.location - self.instantWarpObject1.location
+
+		# Convert from Blender space to SM64 space
+		ret = Vector()
+		ret.x = int(difference.x * bpy.context.scene.blenderF3DScale)
+		ret.y = int(difference.z * bpy.context.scene.blenderF3DScale)
+		ret.z = int(-difference.y * bpy.context.scene.blenderF3DScale)
```

Comment:
prevent rounding errors this way
```suggestion
		ret.x = int(round(difference.x * bpy.context.scene.blenderF3DScale))
		ret.y = int(round(difference.z * bpy.context.scene.blenderF3DScale))
		ret.z = int(round(-difference.y * bpy.context.scene.blenderF3DScale))
```